frontend/src/pages/RunDetailsV2.tsx

```diff
@@ -98,6 +102,18 @@ export function RunDetailsV2(props: RunDetailsV2Props) {
     return 'unknown';
   };
 
+  const { isSuccess: isTemplatePullSuccessFromRun, data: templateString } = useQuery<string, Error>(
+    [runDetail],
+    async () => {
+      if (!runDetail) {
+        throw new Error('No run detail!');
```

Comment:
Do we really need `useQuery`? `runDetail` comes from `RunDetailsV2Props` and I don't see this piece of logic requires making network request.